plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -409,7 +410,49 @@ defineBuiltinTerms = do
             PLC.HeadList -> defineBuiltinInl 'Builtins.head
             PLC.TailList -> defineBuiltinInl 'Builtins.tail
             PLC.ChooseList -> defineBuiltinInl 'Builtins.chooseList
-            PLC.CaseList -> defineBuiltinInl 'Builtins.caseList'
+            PLC.CaseList -> defineBuiltinTerm annMayInline 'Builtins.caseList' $
+                -- > /\a r ->
+                -- >   \(z : r) (f : a -> list a -> r) (xs : list a) ->
+                -- >     chooseList
+                -- >       {a}
+                -- >       {all dead. r}
+                -- >       xs
+                -- >       (/\dead -> z)
+                -- >       (/\dead -> f (headList {a} xs) (tailList {a} xs))
+                -- >       {r}
+                fmap (const annMayInline) . runQuote $ do
```

Comment:
Why not just revert the changes to `PlutusTx.Builtins` and `PlutusTx.Builtins.Internal`? This is more complex and harder to follow.

Reply:
BTW, note that I didn't just write it down from scratch like a maniac, I copy-pasted definitions from `PlutusCore.StdLib.Data.List` and `PlutusCore.StdLib.Data.Data` and tweaked them just a little to get the right argument ordering and `all dead. <...>` instead of `() -> <...>`.